raiden/tests/utils/eth_node.py

```diff
@@ -16,18 +16,19 @@
 from raiden.tests.utils.genesis import GENESIS_STUB, PARITY_CHAIN_SPEC_STUB
 from raiden.utils import privatekey_to_address, privatekey_to_publickey
 from raiden.utils.http import JSONRPCExecutor
-from raiden.utils.typing import Any, Dict, List, NamedTuple
+from raiden.utils.typing import Address, Any, ChainID, Dict, List, NamedTuple, Port, PrivateKey
 
 log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
 
 
+Command = List[str]
 _GETH_VERBOSITY_LEVEL = {"error": 1, "warn": 2, "info": 3, "debug": 4}
 
 
 class EthNodeDescription(NamedTuple):
-    private_key: bytes
-    rpc_port: int
-    p2p_port: int
+    private_key: PrivateKey
+    rpc_port: Port
+    p2p_port: Port
```

Comment:
This type makes little sense IMO - we already know it's a port (`rpc_port` is pretty clear about this). In addition, I would even argue this confuses more than it helps. Let me explain:

Anyone who has ever configured a socket, or made a request to a service on the local network knows that ports are of type `int`. 
Looking at this, I ask myself: why isn't this an `int` as well? Is `Port` a special type?
I would have to double check this isn't the case. Oh. It's an `int` with a fancy name. Hm. Was that really necessary?

I believe you had good intentions - afterall at first glance this *looks* more explicit/prettier. But I would argue it isn't.

A similar issue is the type `PrivateKey` - unless this a special sub-class of `bytes` or a direct result of a function to generate private keys (I haven't checked this - if it **is**, this type is fine), I would advise against using this type here.

Reply:
You are completely missing the point of a type system. As you said, you know that a port is an int, so none of this should have been a surprise to you, however the type checker is not human and needs help, so a new type is introduced allowing the type system to know that an `int` is a port number and not an array index.

Reply:
> You are completely missing the point of a type system. As you said, you know that a port is an int, so none of this should have been a surprise to you, however the type checker is not human and needs help, so a new type is introduced allowing the type system to know that an `int` is a port number and not an array index.

I have not considered this aspect indeed. Thank you for pointing it out. To my mind such strong enforcing of types in python is counter-intuitive, python being duck-typed, but I can see how this makes sense for the project.

Reply:
> To my mind such strong enforcing of types in python is counter-intuitive, python being duck-typed,

Duck typing can be useful, however for the size of the codebase it's terribly hard to know if the correct type is being used. This had been the cause of bugs in the past, and tests do miss it even with high coverage. It's probably unpythonic for some definitions, but it's probably better to be on the safe side then on the err-side.